sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityChecker.scala

```diff
@@ -103,10 +130,85 @@ class StateSchemaCompatibilityChecker(
     }
   }
 
+  def validateAndMaybeEvolveSchema(
```

Comment:
If we are going to replace all calls from check to this one, please ensure that check method is removed.

Reply:
Done